src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -545,7 +558,7 @@ private void OnAsyncTaskStateChanged(DynamoScheduler sender, TaskStateChangedEve
         /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
         /// </summary>
         /// <filterpriority>2</filterpriority>
-        public void Dispose()
+        public virtual void Dispose()
```

Comment:
I don't see why this needs to be made `virtual` (there's no changes for `RevitDynamoModel`). The derived class can completely bypass this by not calling `base.Dispose`, which is dangerous. If this really needs to be `virtual`, then I would recommend the following pattern:

```csharp
public void Dispose()
{
    DisposeCore();
    
    // Do the stuff we do today. The following clean-up codes are guaranteed 
    // to be executed even if derived class does not call base.DisposeCore().
    // 
    LibraryServices.Dispose();
    LibraryServices.LibraryManagementCore.__TempCoreHostForRefactoring.Cleanup();
    Logger.Dispose();
    
    // Bla bla bla...
}

protected virtual void DisposeCore()
{
    // Does nothing here...
}
```